pkg/cmd/roachtest/tests/backup_fixtures.go

```diff
@@ -8,217 +8,219 @@ package tests
-	if sbs.ignoreExistingBackups {
-		cmd = cmd + ",ignore_existing_backups"
-	}
-	return cmd
+// LargeFixture is a TPCC fixture sized so that it is a tight fit in 3 nodes
+// with the maximum supported node density of 8 TiB storage per node. If the
+// node storage density increases, then the size of this fixture should be
+// increased.
+var LargeFixture = TpccFixture{
+	Name:                   "tpcc-300k",
+	ImportWarehouses:       300000,
+	WorkloadWarehouses:     10000,
+	IncrementalChainLength: 400,
+	RestoredSizeEstimate:   "20TiB",
 }
 
 type backupFixtureSpecs struct {
 	// hardware specifies the roachprod specs to create the scheduledBackupSpecs fixture on.
 	hardware hardwareSpecs
 
-	// scheduledBackupSpecs specifies the scheduled scheduledBackupSpecs fixture which will be created.
-	scheduledBackupSpecs scheduledBackupSpecs
-
-	// initWorkloadViaRestore, if specified, initializes the cluster via restore
-	// of an older fixture. The fields specified here will override any fields
-	// specified in the scheduledBackupSpecs field above.
-	initWorkloadViaRestore *restoreSpecs
+	fixture TpccFixture
 
 	timeout time.Duration
 	// A no-op, used only to set larger timeouts due roachtests limiting timeouts based on the suite
 	suites registry.SuiteSet
 
-	testName string
+	clouds []spec.Cloud
 
 	// If non-empty, the test will be skipped with the supplied reason.
 	skip string
 }
 
-func (bf *backupFixtureSpecs) initTestName() {
-	bf.testName = fmt.Sprintf("backupFixture/%s/revision-history=%t/%s", bf.scheduledBackupSpecs.workload.String(), !bf.scheduledBackupSpecs.nonRevisionHistory, bf.scheduledBackupSpecs.cloud)
+const scheduleLabel = "tpcc_backup"
+
+// fixtureFromMasterVersion should be used in the backupSpecs version field to
+// create a fixture using the bleeding edge of master. In the backup fixture
+// path on external storage, the {version} subdirectory will be equal to this
+// value.
+const fixtureFromMasterVersion = "latest"
+
+type scheduledBackupSpecs struct {
+	backupSpecs
+	// ignoreExistingBackups if set to true, will allow a new backup chain
+	// to get written to an already existing backup collection.
+	ignoreExistingBackups    bool
+	incrementalBackupCrontab string
 }
 
-func makeBackupDriver(t test.Test, c cluster.Cluster, sp backupFixtureSpecs) backupDriver {
-	return backupDriver{
-		t:  t,
-		c:  c,
-		sp: sp,
-	}
+func CreateScheduleStatement(uri url.URL) string {
+	// This backup schedule will first run a full backup immediately and then the
+	// incremental backups at the given incrementalBackupCrontab cadence until
+	// the user cancels the backup schedules. To ensure that only one full backup
+	// chain gets created, schedule the full back up on backup will get created
+	// on Sunday at Midnight ;)
+	statement := fmt.Sprintf(
+		`CREATE SCHEDULE IF NOT EXISTS "%s"
+FOR BACKUP DATABASE tpcc
+INTO '%s'
+RECURRING '* * * * *'
+FULL BACKUP '@weekly'
+WITH SCHEDULE OPTIONS first_run = 'now', ignore_existing_backups;
```

Comment:
why is the `ignore_existing_backups` option necessary?

Reply:
Fixed. An early version of this PR didn't have the fixture registry and tried to use SQL to inspect and manage backups in a single shared directory. The pain of trying to clean up leaked fixtures in a single shared director is what lead me to the current fixture registry implementation.